metricbeat/module/http/json/json.go

```diff
@@ -93,37 +96,18 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 		http:            http,
 		requestEnabled:  config.RequestEnabled,
 		responseEnabled: config.ResponseEnabled,
+		responseIsArray: config.ResponseIsArray,
 		deDotEnabled:    config.DeDotEnabled,
 	}, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right format
-// It returns the event which is then forward to the output. In case of an error, a
-// descriptive error must be returned.
-func (m *MetricSet) Fetch() (common.MapStr, error) {
-	response, err := m.http.FetchResponse()
-	if err != nil {
-		return nil, err
-	}
-	defer response.Body.Close()
-
-	var jsonBody map[string]interface{}
-	var event map[string]interface{}
-
-	body, err := ioutil.ReadAll(response.Body)
-	if err != nil {
-		return nil, err
-	}
-
-	err = json.Unmarshal(body, &jsonBody)
-	if err != nil {
-		return nil, err
-	}
+func (m *MetricSet) processBody(response *http.Response, jsonBody interface{}) common.MapStr {
+	var event common.MapStr
 
 	if m.deDotEnabled {
-		event = common.DeDotJSON(jsonBody).(map[string]interface{})
+		event = common.DeDotJSON(jsonBody).(common.MapStr)
 	} else {
-		event = jsonBody
+		event = jsonBody.(common.MapStr)
```

Comment:
We probably should do some type conversion checks on both lines?

Reply:
Sorry, I don't understand what you mean by 'both lines'. Elaborate please?

Reply:
Sorry, I meant on line 106 and 108 as both to a type converstion to `common.MapStr`.

Reply:
Did assertion because `common.DeDotJSON` accepts and emits `interface{}`. Type conversion would fail.

If types are mismatched, line 160 and 170 should error out.

Reply:
One more miscommunication on my end. I meant type assertions. I was looking for:

```
event, ok = jsonBody.(common.MapStr)
if !ok { ... }
```

If for whatever reason the assertion does not work, we would have a panic otherwise. At the same time I see it should not happen here.